libs/community/tests/unit_tests/document_loaders/test_gitbook.py

```diff
@@ -0,0 +1,204 @@
+    # Test that the loader uses the default sitemap URL when load_all_paths=True
+    loader = GitbookLoader(web_page="https://example.com", load_all_paths=True)
+
+    # Check that the web_path was set to the default sitemap URL
+    assert loader.web_paths[0] == "https://example.com/sitemap.xml"
+
+
+@patch("langchain_community.document_loaders.web_base.requests.get")
+def test_init_with_custom_sitemap(mock_get: MagicMock) -> None:
+    # Test that the loader uses the provided sitemap URL when specified
+    custom_sitemap = "https://example.com/sitemap-pages.xml"
+    loader = GitbookLoader(
+        web_page="https://example.com",
+        load_all_paths=True,
+        sitemap_url=custom_sitemap,
+    )
+
+    # Check that the web_path was set to the custom sitemap URL
+    assert loader.web_paths[0] == custom_sitemap
+
+
+@patch("langchain_community.document_loaders.gitbook.GitbookLoader.scrape")
+@patch("langchain_community.document_loaders.gitbook.GitbookLoader.scrape_all")
+def test_lazy_load_with_custom_sitemap(
+    mock_scrape_all: MagicMock,
+    mock_scrape: MagicMock,
+    mock_soups: Tuple[BeautifulSoup, BeautifulSoup],
+) -> None:
+    # Setup the mocks
+    mock_sitemap_soup, mock_page_soup = mock_soups
+    mock_scrape.return_value = mock_sitemap_soup
+    mock_scrape_all.return_value = [
+        mock_page_soup,
+        mock_page_soup,
+        mock_page_soup,
+    ]
+
+    # Create loader with custom sitemap URL
+    loader = GitbookLoader(
+        web_page="https://example.com",
+        load_all_paths=True,
+        sitemap_url="https://example.com/sitemap-pages.xml",
+    )
+
+    # Get the documents
+    docs = list(loader.lazy_load())
+
+    # Check that we got docs for each path in the sitemap
+    assert len(docs) == 3
+    for doc in docs:
+        assert doc.metadata["title"] == "Test Page"
+        assert "This is test content." in doc.page_content
+
+
+@patch("langchain_community.document_loaders.web_base.requests.get")
+def test_with_single_page(mock_get: MagicMock) -> None:
+    # Test loading a single page (load_all_paths=False)
+    loader = GitbookLoader(web_page="https://example.com/page", load_all_paths=False)
+
+    # Check that sitemap URL logic was not applied
+    assert loader.web_paths[0] == "https://example.com/page"
+
+
+@patch("langchain_community.document_loaders.gitbook.GitbookLoader.scrape")
+def test_get_paths_extraction(
+    mock_scrape: MagicMock, mock_soups: Tuple[BeautifulSoup, BeautifulSoup]
+) -> None:
+    # Test that _get_paths correctly extracts paths from sitemap
+    mock_sitemap_soup, _ = mock_soups
+    mock_scrape.return_value = mock_sitemap_soup
+
+    loader = GitbookLoader(web_page="https://example.com", load_all_paths=True)
+
+    soup_info = loader.scrape()
+    paths = loader._get_paths(soup_info)
+
+    # Check that paths were extracted correctly
+    assert len(paths) == 3
+    assert paths == ["/page1", "/page2", "/page3"]
+
+
+@patch("requests.get")
```

Comment:
Thanks!

@andrasfe I don't know if you're familiar with the responses library, but it's generally a **much** cleaner way to patch requests.get / patch / post if you're comparing it to using the built in patch methods.

Going to merge as is (not a blocker). But if you have time to re-write w/ responses will be appreciated. It'll make the code much less brittle!

---

I've observed that the usage of `@patch` w/ strings specifying the patched code introduces more technical debt sometimes than it solves. patch object is slightly better since it's easier to catch namespacing changes, but for network requests `responses` is better in virtually all cases